huaweicloud/services/iam/data_source_huaweicloud_identity_group.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+func DataSourceIdentityGroup() *schema.Resource {
+	return &schema.Resource{
+		Read: DataSourceIdentityGroupV3Read,
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"domain_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"users": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"domain_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"password_expires_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"password_status": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"password_strength": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func DataSourceIdentityGroupV3Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	identityClient, err := config.IdentityV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.Errorf("Error creating HuaweiCloud identity client: %s", err)
+	}
+
+	listOpts := groups.ListOpts{
+		Name: d.Get("name").(string),
+	}
+
+	allPages, err := groups.List(identityClient, listOpts).AllPages()
+	if err != nil {
+		return fmtp.Errorf("Unable to query groups: %s", err)
+	}
+
+	allGroups, err := groups.ExtractGroups(allPages)
+
+	if err != nil {
+		return fmtp.Errorf("Unable to retrieve groups: %s", err)
+	}
+
+	conditions := map[string]interface{}{}
+
+	if v, ok := d.GetOk("id"); ok {
+		conditions["id"] = v.(string)
+	}
+	if v, ok := d.GetOk("description"); ok {
+		conditions["description"] = v.(string)
+	}
+
+	var foundGroups []groups.Group
+
+	for _, group := range allGroups {
+		if groupsFilter(group, conditions) {
+			foundGroups = append(foundGroups, group)
+		}
+	}
+
+	if len(foundGroups) < 1 {
+		return fmtp.Errorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
+	}
+
+	if len(foundGroups) > 1 {
+		logp.Printf("[DEBUG] Multiple results found: %#v", foundGroups)
+		return fmtp.Errorf("Your query returned more than one result. " +
+			"Please try a more specific search criteria.")
+	}
+
+	group := foundGroups[0]
+
+	d.SetId(group.ID)
+	d.Set("domain_id", group.DomainID)
+	d.Set("name", group.Name)
```

Comment:
`domain_id` and `name` should not be allowed to be empty, suggest to use `multierror` to set values.

Reply:
fixed